Loop/Managers/LoopDataManager.swift

```diff
@@ -657,6 +654,40 @@ extension LoopDataManager {
             }
         }
     }
+    
+    /// Logs a new external bolus insulin dose in the DoseStore and HealthKit
+    ///
+    /// - Parameters:
+    ///   - startDate: The date the dose was started at.
+    ///   - value: The number of Units in the dose.
+    ///   - insulinModel: The type of insulin model that should be used for the dose.
+    func logOutsideInsulinDose(startDate: Date, units: Double, insulinModelSetting: InsulinModelSettings? = nil) {
```

Comment:
Created this function so there's a separate pathway to log doses vs administer a bolus dose